### DISCUSSION TITLE
Configuring vscode formatter in ojs cells with Quarto

### DISCUSSION THREAD

**Discussion by boshek** [quarto-dev/quarto-cli#7666] [Q&A]
### Description

Has any worked out how to get vscode formatting tools to work with Quarto and ojs chunks? For example if I have the following in my settings:

```json
    "[javascript]":  {
        "editor.defaultFormatter": "esbenp.prettier-vscode"
	},
```

I can format this javascript code _if_ it is in a javascript file:
![image](https://github.com/quarto-dev/quarto-cli/assets/18472598/66d8cf5c-168b-47f9-b5a0-95595a895e3d)

If I try to add a formatter to quarto like so:

```json
    "[quarto]":  {
        "editor.defaultFormatter": "esbenp.prettier-vscode"
	},
```

I get the following error message when I try to format the same code but within an ojs chunk with a qmd:
![image](https://github.com/quarto-dev/quarto-cli/assets/18472598/84b755df-9a63-4a59-ba9f-4eccb0768346)

If I click "Configure..." and then choose quarto as the default it automatically changes the setting to:

```json
    "[quarto]":  {
        "editor.defaultFormatter": "quarto.quarto"
	},
```
and the code in the ojs chunk isn't formatted nicely. Has anyone else encountered this for ojs chunks? Formatting code this way works out of the box for python and R (at least for general formatting) but I am having trouble making it work for ojs chunks. Just posting in case anyone has encounted this before.

**Comment by mcanouil** [comment]
Quarto relies on specific language extensions to format code.
Unfortunately, I do not know  the internals of the Quarto extension for VSCode (https://github.com/quarto-dev/quarto).

@jjallaire might be the best to answer here.

**Comment by zmaril** [comment]
This is frustrating for me as well, I wish it would work better.

**Reply by boshek** [reply]
I've realized that for, the actual tricky part of this is that I don't know how to lint a quarto doc now.

**Reply by zmaril** [reply]
Me either, mine are turning into a mess right now.

**Reply by mcanouil** [reply]
The issue is not Quarto.

Python/JavaScript/etc. linters don't lint markdown code blocks.
Only R linter parses markdown for R code blocks.

**Reply by boshek** [reply]
Totally agree that this no fault of Quarto's. But I also think it is a problem worthy of discussion.

**Reply by boshek** [reply]
Just found this at least for python: https://github.com/shafayetShafee/black-formatter
